### PR TITLE
Add 1.9 Support options for work with VSF Mage Indexer defaults

### PR DESCRIPTION
**Pull request by tonyisworking**
Added changes for VSF 1.9 Urls

### CODE REVIEW

**Comment by tonyisworking** [comment]
Needs some tests to be added

**Comment by Tjitse-E** [comment]
Thanks for your contribution!

File: Console/RunGenerateSitemap.php [hunk 1]
```diff
@@ -0,0 +1,34 @@
+<?php
+namespace Vendic\VueStorefrontSitemap\Console;
+
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class RunGenerateSitemap extends Command
+{
+    protected $generateSitemapCron;
```
Comment (Tjitse-E) [inline Console/RunGenerateSitemap.php]: Can you add a typehint and enter between the constructor and property?

File: Console/RunGenerateSitemap.php [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    public function __construct(
+        \Vendic\VueStorefrontSitemap\Cron\GenerateSitemap $generateSitemapCron
```
Comment (Tjitse-E) [inline Console/RunGenerateSitemap.php]: Can you import the class?

File: Console/RunGenerateSitemap.php [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        parent::configure();
+    }
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        try {
+            $this->generateSitemapCron->execute();
+            $output->writeln("VSF Sitemap Generated");
+        } catch (\Exception $e) {
+            $output->writeln($e->getMessage());
+        } catch (Throwable $e) {
```
Comment (Tjitse-E) [inline Console/RunGenerateSitemap.php]: This class `Throwable` seems to be undefined.

File: Model/CategoryCollection.php [hunk 4]
```diff
@@ -28,10 +28,18 @@ public function __construct(
...
+        ->create()
+        ->addFieldToSelect('url_key')
+        ->addFieldToFilter('url_key', ['neq' => null])
+        ->addFieldToSelect('url_path')
+        ->addFieldToFilter('url_path', ['neq' => null])
+        ->addFieldToSelect('is_active')
+        ->addFieldToSelect('include_in_menu')
+        ->addFieldToSelect('level')
+        ->addFieldToFilter('is_active', ['eq' => 1])
+        ->addFieldToFilter('include_in_menu', ['eq' => 1])
```
Comment (Tjitse-E) [inline Model/CategoryCollection.php]: Why do we  need to add an `include_in_menu` filter? Categories in the sitemap, but not in the menu shouldn't be impossible right?

Comment (tonyisworking) [reply Model/CategoryCollection.php]: That is true. I'll update.

**Comment by tonyisworking** [comment]
Updated with requested changes

**Pull request merged**